### DISCUSSION TITLE
Jwt data type

### DISCUSSION THREAD

**Discussion by Tarang** [supabase/supabase#857] [Questions] [answered]
I am writing my own row level security policy. I am trying to write my own `auth.uid` function. I was wondering where I can find the code for how jwt are parsed, and specifically the data type on them. So if my jwt has a field (which is an array), i can parse it properly.

**Comment by steve-chavez** [comment, marked as answer]
This is what `auth.uid` does:

```sql
select nullif(current_setting('request.jwt.claim.sub', true), '')::uuid;
```
(You can get the above by doing `select prosrc from pg_proc where proname = 'uid';` on the Query Editor)

> So if my jwt has a field (which is an array)

You can get any custom claim with `current_setting('request.jwt.claim.<custom>', ...`, as done above. You can even cast it to json or to a postgres array.

**Reply by Tarang** [reply]
@steve-chavez I've got to that part but I dont know what the type is of `current_setting('request.jwt.claim.<custom>', true)` so its making it very hard to cast to json. Especially if my `<custom>` is an array such as `["a","b"]`, i can't cast it. Is it possible to cast it to json?

**Reply by steve-chavez** [reply]
> Is it possible to cast it to json?

Yes, this works:
```sql
select '["a", "b"]'::json;
-- ["a","b"]
```

So you could do:

```sql
select current_setting('request.jwt.claim.<custom>', true)::json;
```
